tests/test_frontier.py

```diff
@@ -230,12 +230,123 @@ def test_resume_job():
     assert len(job.starts_and_stops) == 3
     assert job.starts_and_stops[2]['start']
     assert job.starts_and_stops[2]['stop']
-    assert job.starts_and_stops[2]['stop'] > job.starts_and_stops[0]['start']
+    assert job.starts_and_stops[2]['stop'] > job.starts_and_stops[2]['start']
     assert site.status == 'FINISHED'
     assert len(site.starts_and_stops) == 3
     assert site.starts_and_stops[2]['start']
     assert site.starts_and_stops[2]['stop']
-    assert site.starts_and_stops[2]['stop'] > site.starts_and_stops[0]['start']
+    assert site.starts_and_stops[2]['stop'] > site.starts_and_stops[2]['start']
+
+    frontier.resume_job(job)
+    site = list(frontier.job_sites(job.id))[0]
+
+    assert job.status == 'ACTIVE'
+    assert len(job.starts_and_stops) == 4
+    assert job.starts_and_stops[3]['start']
+    assert job.starts_and_stops[3]['stop'] is None
+    assert site.status == 'ACTIVE'
+    assert len(site.starts_and_stops) == 4
+    assert site.starts_and_stops[3]['start']
+    assert site.starts_and_stops[3]['stop'] is None
+
+    # simulate a job stop request
+    job_conf = {'seeds': [{'url': 'http://example.com/'}, {'url': 'http://example_2.com/'}]}
+    job = brozzler.new_job(frontier, job_conf)
+    assert len(list(frontier.job_sites(job.id))) == 2
+    site1 = list(frontier.job_sites(job.id))[0]
+    site2 = list(frontier.job_sites(job.id))[1]
+
+    job.stop_requested = datetime.datetime.utcnow().replace(tzinfo=doublethink.UTC)
+    job.save()
+
+    # should raise a CrawlStopped
+    with pytest.raises(brozzler.CrawlStopped):
+        frontier.honor_stop_request(site1)
+
+    frontier.finished(site1, 'FINISHED_STOP_REQUESTED')
+    frontier.finished(site2, 'FINISHED_STOP_REQUESTED')
+    job.refresh()
+
+    assert job.status == 'FINISHED'
+    assert job.stop_requested
+    assert len(job.starts_and_stops) == 1
+    assert job.starts_and_stops[0]['start']
+    assert job.starts_and_stops[0]['stop']
+    assert job.starts_and_stops[0]['stop'] > job.starts_and_stops[0]['start']
+    assert site1.status == 'FINISHED_STOP_REQUESTED'
+    assert site2.status == 'FINISHED_STOP_REQUESTED'
+    assert len(site1.starts_and_stops) == 1
+    assert len(site2.starts_and_stops) == 1
+    assert site1.starts_and_stops[0]['start']
+    assert site1.starts_and_stops[0]['stop']
+    assert site1.starts_and_stops[0]['stop'] > site.starts_and_stops[0]['start']
+    assert site2.starts_and_stops[0]['start']
+    assert site2.starts_and_stops[0]['stop']
+    assert site2.starts_and_stops[0]['stop'] > site.starts_and_stops[0]['start']
+
+    # simulate job resume after a stop request
+    frontier.resume_job(job)
+    site1 = list(frontier.job_sites(job.id))[0]
+    site2 = list(frontier.job_sites(job.id))[0]
```

Comment:
Looks like you made the same kind of mistake here that I had made and you fixed above! :) I.e. I think it should be
```
    site2 = list(frontier.job_sites(job.id))[1]
```

Reply:
xD